nncf/openvino/graph/layer_attributes.py

```diff
@@ -67,9 +67,97 @@ def get_const_port_ids(self) -> List[int]:
+    OVGroupConvolutionBackpropDataMetatype,
+]
+
+
+def get_conv_weights_layout_from_node(node: NNCFNode) -> List[OVLayoutElem]:
+    """
+    Calculates weights layout for a target convolution node.
+
+    :param node: Target convolution node.
+    :return: Target convolution Node weights layout.
+    """
+    layer_attributes = node.layer_attributes
+    port_id = _get_constant_port_id_from_layer_attributes(layer_attributes)
+    return get_conv_weights_layout(
+        ov_metatype=node.metatype, weights_shape=layer_attributes.constant_attributes[port_id]["shape"]
+    )
+
+
+def get_linear_weights_layout_from_node(node: NNCFNode) -> List[OVLayoutElem]:
+    """
+    Calculates weights layout for a target linear node.
+
+    :param node: Target linear node.
+    :return: Target linear Node weight layout.
+    """
+    layer_attributes = node.layer_attributes
+    port_id = _get_constant_port_id_from_layer_attributes(layer_attributes)
+    constant_layer_attrs = layer_attributes.constant_attributes[port_id]
+    return get_linear_weights_layout(
+        weights_shape=constant_layer_attrs["shape"],
+        transpose=constant_layer_attrs["transpose"],
+        port_id=port_id,
+    )
+
+
+def _get_constant_port_id_from_layer_attributes(layer_attributes: OVLayerAttributes) -> int:
+    """
+    Returns constant ports id for convolutional and linear ops layer attributes.
+
+    :param layer_attributes: Target convolutional/linear layer op layer attributes.
+    :return: Constant port id for the target convolutional/linear model.
+    """
+    port_ids = list(layer_attributes.constant_attributes.keys())
+    assert len(port_ids) == 1
+    return port_ids[0]
+
+
+def get_conv_weights_layout(ov_metatype: OVOpMetatype, weights_shape: Tuple[int, ...]) -> List[OVLayoutElem]:
+    """
+    Calculates weights layout for a target convolution node.
+
+    :param ov_metatype: Target convolution node OpenVINO metatype.
+    :param weights_shape: Shape of the target convolution node weight.
+    :return: Target convolution node weights layout.
+    """
+    weights_layout = ov_metatype.const_layout
+    kernel_size = weights_shape[len(weights_layout) :]
+    weights_layout += [OVLayoutElem.SPATIAL] * len(kernel_size)
+    return tuple(weights_layout)
+
+
+def get_linear_weights_layout(weights_shape: Tuple[int, ...], transpose: bool, port_id: int) -> List[OVLayoutElem]:
+    """
+    Calculates weights layout for a target linear node.
+
+    :param weights_shape: Shape of the target linear node weight.
+    :param port_id: Port id of the target liner node weights.
+    :return: Target linear node weight layout.
+    """
+    weights_layout = [OVLayoutElem.SPATIAL] * (len(weights_shape) - 2)
+    if len(weights_shape) > 1:
+        if (transpose and port_id == 0) or (not transpose and port_id == 1):
+            weights_layout += [OVLayoutElem.C_IN, OVLayoutElem.C_OUT]
+        else:
+            weights_layout += [OVLayoutElem.C_OUT, OVLayoutElem.C_IN]
```

Comment:
Minor. We have an example in the `calculate_port_based_channel_axis` method that looks more reliable than a bunch of the `and/or` conditions.

Reply:
I would leave it as it is, looks like this if is covering all possible cases for the matmul operation